.github/workflows/ci_build_android.yml

```diff
@@ -0,0 +1,50 @@
+name: Build Android library
+
+on:
+  workflow_call:
+    outputs:
+      artifact-dir-base:
+        description: The base name of artifact directory (without the target suffix)
+        value: ${{ jobs.build-android-libs.outputs.artifact-dir-base }}
+
+jobs:
+  build-android-libs:
+    runs-on: ubuntu-20.04
+    timeout-minutes: 35
+    strategy:
+      matrix:
+        target: ["arm64-v8a", "x86_64"]
+    env:
+      ARTIFACT_DIR_BASE: ${{ github.job }}
+    outputs:
+      artifact-dir-base: ${{ env.ARTIFACT_DIR_BASE }}
+    steps:
+      - uses: actions/checkout@ec3a7ce113134d7a93b817d10a8272cb61118579 # v2.4.0
+
+      - name: Setup the rust CI.
+        uses: ./.github/actions/setup-job
+        with:
+          rust: true
+          dart: true
+          android: true
+          rust-target: "aarch64-linux-android x86_64-linux-android"
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Build Android library
+        run: |
+          export PATH="$(pwd)/cargo-installs/bin:${PATH}"
```

Comment:
why do we need that?

Reply:
Because otherwise it doesn't find `cargo-ndk`

Reply:
that is weird because we never had to do this before 
also cargo-ndk shound't be install in `cargo-installs/bin` but in `~/.cargo/bin/`
`cargo-installs/bin` that is only for `async-bindgen-gen-dart`
can you try a run without that line please

Reply:
Breaks [here](https://github.com/xaynetwork/xayn_discovery_engine/runs/5214694142?check_suite_focus=true#step:4:25)

Reply:
ah i think it because of `CARGO_INSTALL_ROOT` in `.env`. that will install cargo-ndk in `cargo-installs/bin`. @acrrd, @rustonaut is `cargo-installs/bin` only there for `async-bindgen-gen-dart` or was there the plan to also use it also for cargo-ndk, cargo sort,... ?

Reply:
I think the `CARGO_INSTALL_ROOT` is not a blocker for now. We can do it in a follow-up pr if we find a better solution. 
to make it a bit cleaner we could move this line to the step where we install cargo-ndk. 
but the line would need to change to: [`echo "${{ github.workspace }}/cargo-installs/bin" >> $GITHUB_PATH`](https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#adding-a-system-path)
wdyt?

Reply:
How is that failing, I mean sure `CARGO_INSTALL_ROOT` does install it somewhere else, but it also makes look `cargo` at that place when running a `cargo ..` sub-command :thinking: 

And `just` always loads `.env` files so the env should match.

Am I missing something?

Reply:
Seems that cargo ignores `CARGO_INSTALL_ROOT` when you try to run an executable :( They also point out adding the path:
```
  Installing cargo-installs/bin/async-bindgen-gen-dart
   Installed package `async-bindgen-gen-dart v0.1.0 (https://github.com/xaynetwork/xayn_async_bindgen.git#fe195d99)` (executable `async-bindgen-gen-dart`)
warning: be sure to add `cargo-installs/bin` to your PATH to be able to run the installed binaries
```

Reply:
Yes, but it's to run them without going through cargo I thought.

I mean what point is there in being able to change the install root if it then is ignored?

Reply:
I think whatever the reasoning may be, `CARGO_INSTALL_ROOT` appears to not automagically also make the sub-command available to cargo, which is why we need to modify the `PATH` for `cargo-ndk` to work. This is necessary because we also can't invoke the `cargo-ndk` binary directly. I will go ahead and merge once the current state passes, then we can think about a different solution later on.